fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1708,6 +1713,7 @@ public Builder setSubspace(@Nonnull Subspace subspace) {
          */
         public Builder setIndexStatePrecondition(@Nonnull IndexStatePrecondition indexStatePrecondition) {
             this.indexStatePrecondition = indexStatePrecondition;
+            this.explicitIndexStatePrecondition = true;
```

Comment:
Rather than keeping a boolean flag around to indicate whether the index state precondition has been set, I think you could have this update the `IndexingPolicy` directly, either by changing the type of `IndexingPolicy` to `IndexingPolicy.Builder` and mutating the index state `DesiredAction`s in place, or by copying the current indexing policy into a new `Builder`, updating the states on the builder, and then building and setting the `IndexingPolicy`. I think that would also allow you to avoid needing to mutate private data of the `IndexingPolicy` in the `build()` method, which we probably want to avoid.

Reply:
Wouldn't changing `setIndexingPolicy(@Nullable final IndexingPolicy indexingPolicy)` to `setIndexingPolicy(@Nullable final IndexingPolicy.Builder builder)` break backward compatibility? 
If so, are you ok with it?

Reply:
Temporary, I've kept them both. 
(This may be changed in the final version of this PR)

Reply:
Got it. Yeah, I think it would be "probably fine" to keep it as just the indexing policy and not the builder, if you want to remove one rather than keep two. Or maybe both is fine. It's somewhat up to you.